### PR TITLE
Feature/basic admin api

### PR DESCRIPTION
**Pull request by parc-jason**
# Description
Add basic api for the submission/attestation administration.

I renamed the *_notes table to *_note for consistency.  Dev Master will need to have the migrations rolled back.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/forms/minesattestations/dataService.js [hunk 1]
```diff
@@ -119,16 +119,134 @@ const dataService = {
...
+      .withGraphFetched('statuses(orderDescending).notes(orderDescending)');
+    return results || [];
+  },
+
+  createSubmissionStatus: async (obj, confirmationId, user) => {
+    if (!obj) {
+      throw Error('Status cannot be created without data');
+    }
+    let trx;
+    let result;
```
Comment (jujaga) [inline app/src/forms/minesattestations/dataService.js:147]: Result can be declared as a const inside the try block as it's only used there on return. You could actually even get away with just return awaiting line 160 with line 162.

File: app/src/forms/minesattestations/dataService.js [hunk 2]
```diff
@@ -119,16 +119,134 @@ const dataService = {
...
+      .modify('orderDescending')
+      .throwIfNotFound();
+  },
+
+  createSubmissionStatusNote: async (obj, confirmationId, statusId, user) => {
+    if (!obj) {
+      throw Error('Note cannot be created without data');
+    }
+    let trx;
+    let result;
```
Comment (jujaga) [inline app/src/forms/minesattestations/dataService.js:186]: Result can be declared as a const inside the try block as it's only used there on return. You could actually even get away with just return awaiting line 195 with line 197.

File: app/src/forms/minesattestations/dataService.js [hunk 3]
```diff
@@ -119,16 +119,134 @@ const dataService = {
...
+      .modify('orderDescending');
+    return results || [];
+  },
 
+  createSubmissionNote: async (obj, confirmationId, user) => {
+    if (!obj) {
+      throw Error('Note cannot be created without data');
+    }
+    let trx;
+    let result;
```
Comment (jujaga) [inline app/src/forms/minesattestations/dataService.js:220]: Result can be declared as a const inside the try block as it's only used there on return. You could actually even get away with just return awaiting line 229 with line 231.

File: app/src/forms/minesattestations/routes.js [hunk 4]
```diff
@@ -14,13 +14,45 @@ routes.post('/', async (req, res, next) => {
...
+routes.get('/submissions', async (req, res, next) => {
+  await controller.allSubmissions(req, res, next);
+});
+
+routes.post('/submissions', async (req, res, next) => {
   await controller.createSubmission(req, res, next);
 });
 
-routes.get('/submission/:confirmationId', async (req, res, next) => {
+routes.put('/submissions/:confirmationId', async (req, res, next) => {
```
Comment (loneil) [inline app/src/forms/minesattestations/routes.js:25]: Don't know if we care, but "Confirmation ID" was generally referring to the shortform of the guid that the user gets. So this is really the "ID" primary key right? Not the "Confirmation ID".
Minor semantic so not concerned but think about whether we want to call the REST identifier for the submission "confirmation Id"? Its the submissionId in the table right?

Comment (parc-jason) [reply app/src/forms/minesattestations/routes.js:25]: yeah, confirmationId is just the short submissionId, but I use that in the API for grabbing a submission.  shorter...  although i guess no one ever sees it... 🤷‍♂️

**Pull request merged**